src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -664,29 +660,26 @@ def repartition(self, axis: Any = None) -> "SnowflakeQueryCompiler":
 
     def default_to_pandas(
         self, pandas_op: Callable, *args: Any, **kwargs: Any
-    ) -> "SnowflakeQueryCompiler":
+    ) -> NoReturn:
         func_name = pandas_op.__name__
 
         # this is coming from Modin's encoding scheme in default.py:build_default_to_pandas
         # encoded as f"<function {cls.OBJECT_TYPE}.{fn_name}>"
         # extract DataFrame operation, following extraction fails if not adhering to above format
         object_type, fn_name = func_name[len("<function ") : -1].split(".")
 
-        # in case it's a column wise or other unspecified operation, use a stored procedure which
-        # will materialize everything as a pandas DataFrame within Snowflake.
-        # Because the Snowflake executor has limited memory, this fallback likely will fail for large datasets.
-        return_query_compiler = StoredProcedureDefault.register(
-            self._modin_frame, pandas_op, args, kwargs
-        )
-        return_query_compiler.snowpark_pandas_api_calls.append(
-            {
-                TelemetryField.NAME.value: f"{object_type}.{fn_name}",
-                SnowparkPandasTelemetryField.IS_FALLBACK.value: True,
-            }
+        # Previously, Snowpark pandas would register a stored procedure that materializes the frame
+        # and performs the native pandas operation. Because this fallback has extremely poor
+        # performance, we now raise NotImplementedError instead.
+        args_str = ", ".join(map(str, args))
+        if args and kwargs:
+            args_str += ", "
+        if kwargs:
+            args_str += ", ".join(f"{key}={value}" for key, value in kwargs.items())
+        ErrorMessage.not_implemented(
+            f"{object_type}.{fn_name}({args_str}) is not yet available in Snowpark pandas API"
```

Comment:
Just to be consistent with what was mentioned on Slack:
```suggestion
            f"Snowpark pandas doesn't yet support {object_type}.{fn_name}({args_str}"
```

Reply:
Changed. Note that since this method call occurs behind a few layers of indirection (for example, `BinaryDefault.register` -> `DataFrameDefault.register` -> `SnowflakeQueryCompiler.default_to_pandas()`) we lose information about which specific parameter caused the error, so a followup PR should replace calls in the QC to `*Default.register` with a direct `ErrorMessage.not_implemented` to call out the specific method/param as needed.

Reply:
Is it worth filing a ticket for that?

Reply:
I think this would be covered by SNOW-1341619, though that ticket itself might need to get split into smaller chunks.

Reply:
Ah okay, cool.